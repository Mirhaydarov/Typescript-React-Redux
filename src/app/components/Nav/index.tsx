// Core
import React, { FC, ReactElement } from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';

// Components
import { SearchBar } from '../SearchBar';

// Route
import { book } from '../../routes/book';

// Styles
import styles from './styles.module.scss';

type RouteParams = {
  push: string;
  pathname: string;
};

type NavProps = RouteComponentProps<RouteParams>;

export const Nav: FC<NavProps> = (props: NavProps): ReactElement => {
  const {
    location: { pathname },
  } = props;

  return (
    <header className={styles.header}>
      <nav className={styles.nav}>
        {pathname === book.root ? null : (
          <>
            <Link className={styles.navLink} to={book.root}>
              Home
            </Link>
          </>
        )}
        <SearchBar />
      </nav>
    </header>
  );
};
