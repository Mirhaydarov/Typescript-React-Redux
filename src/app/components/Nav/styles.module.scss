$header-bg-color: #89a8ff;
$nav-bg-color: #89a8ff;
$btn-bg-color: #f8f8ff;
$btn-border-color: #f8f8ff;
$btn-font-color: #000;
$btn-font-color--hover: #f8f8ff;
$header-border-color: #000;
$header-shadow-color: #000;

.header {
  height: 90px;
  background-color: $header-bg-color;
  border-bottom: 1px solid $header-border-color;
  box-shadow: 0px 1px 3px $header-shadow-color;
}

.nav {
  display: grid;
  grid-template-columns: repeat(2, max-content);
  grid-gap: 15px;
  align-items: center;
  justify-content: center;
  height: 100%;

  &__link {
    padding: .5rem 1rem;
    background-color: $btn-bg-color;
    color: $btn-font-color;
    border: 2px solid $btn-border-color;
    border-radius: 3px;
    font-size: 1rem;
    cursor: pointer;
    transition: .4s ease-in-out;
    outline: none;
    text-decoration: none;
    &:hover {
      background-color: darken($nav-bg-color, 15);
      border: 2px solid $btn-border-color;
      color: $btn-font-color--hover;
    }
    &:focus {
      border: 2px solid darken($nav-bg-color, 15);
    }
  }
}