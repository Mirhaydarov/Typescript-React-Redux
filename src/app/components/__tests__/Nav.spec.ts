// Core
import { createMemoryHistory } from 'history';
import { match } from 'react-router-dom';

// Components
import { Nav } from '../Nav';

// Instruments
import {
  makeMountRender,
  reduxify,
  store,
} from '../../../test-utils';

const history = createMemoryHistory();
const path = `/profile/:userName`;

const match: match<{ userName: string }> = {
  isExact: false,
  path,
  url: path.replace(':userName', 'leanne-graham'),
  params: { userName: 'leanne-graham' },
};

const routeComponentPropsMock = (pathname: string) => ({
  location: {
    pathname,
    hash: '',
    key: '',
    search: '',
    state: {},
  },
  history,
  match,
});

describe('<Nav />', () => {
  it('Don\'t render button Home if pathname is "/"', () => {
    const props = routeComponentPropsMock('/');
    const wrapper = makeMountRender(reduxify(Nav, store, props));

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Render button Home if pathname different then "/"', () => {
    const props = routeComponentPropsMock('/profile');
    const wrapper = makeMountRender(reduxify(Nav, store, props));

    expect(wrapper.html()).toMatchSnapshot();
  });
});
