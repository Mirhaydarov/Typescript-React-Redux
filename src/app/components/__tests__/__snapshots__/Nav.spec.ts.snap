// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> Don't render button Home if pathname is "/" 1`] = `
<header class="header">
  <nav class="nav">
    <input class="searchBar"
           type="search"
           placeholder="Search people by name..."
           value
    >
  </nav>
</header>
`;

exports[`<Nav /> Render button Home if pathname different then "/" 1`] = `
<header class="header">
  <nav class="nav">
    <a class="navLink"
       href="/"
    >
      Home
    </a>
    <input class="searchBar"
           type="search"
           placeholder="Search people by name..."
           value
    >
  </nav>
</header>
`;
